kconfig: Global whitespace/consistency cleanup

ulfalizer opened the pull request:
Clean up space errors and use a consistent style throughout the Kconfig
files. This makes reading the Kconfig files more distraction-free, helps
with grepping, and encourages the same style getting copied around
everywhere (meaning another pass hopefully won't be needed).

Go for the most common style:

 - Indent properties with a single tab, including for choices.

   Properties on choices work exactly the same syntactically as
   properties on symbols, so not sure how the no-indentation thing
   happened.

 - Indent help texts with a tab followed by two spaces

 - Put a space between 'config' and the symbol name, not a tab. This
   also helps when grepping for definitions.

 - Do '# A comment' instead of '#A comment'

Some help texts were reflowed to 79 columns with 'gq' in Vim as well,
though not all, because I was afraid I'd accidentally mess up
formatting.

I tweaked Kconfiglib a bit to find most of the stuff.

ulfalizer commented:
And yeah, this stomps a bit on the history for some symbols, but there's almost always handles left for `git blame`.

Once stuff looks the same everywhere, I think things will keep themselves relatively consistent by themselves.

dcpleung commented on drivers/kscan/Kconfig.xec:
```diff
@@ -18,32 +18,32 @@ config KSCAN_XEC_COLUMN_SIZE
-	 Adjust the value to your keyboard columns. The maximum
-	 column size for the Microchip XEC family is 18 (from 0 to 17).
+	  Adjust the value to your keyboard columns. The maximum
+	  colum size for the Microchip XEC family is 18 (from 0 to 17).
```
typo?

ulfalizer replied:
Fixed. Wonder how that happened...

dcpleung reviewed (approved):
LGTM.

galak commented on drivers/video/Kconfig.sw_generator:
```diff
@@ -6,7 +6,7 @@
-        bool "Video Software Generator"
-        help
-          Enable video pattern generator (for testing purpose).
+config VIDEO_SW_GENERATOR
```
was this meant to go from menuconfig -> config?  If so maybe doing that in a separate commit.

galak commented on drivers/video/Kconfig.mt9m114:
```diff
@@ -6,8 +6,8 @@
-        depends on I2C
-        help
-          Enable driver for MT9M114 CMOS digital image sensor device.
+config VIDEO_MT9M114
```
was this meant to go from menuconfig -> config? If so maybe doing that in a separate commit.

ulfalizer replied:
There's a separate commit at https://github.com/zephyrproject-rtos/zephyr/pull/20165. Was an attempt to avoid a clash with it.

ulfalizer replied:
Removed

mbolivar reviewed (approved):
+1 led_strip changes

ulfalizer committed:
kconfig: Global whitespace/consistency cleanup

Clean up space errors and use a consistent style throughout the Kconfig
files. This makes reading the Kconfig files more distraction-free, helps
with grepping, and encourages the same style getting copied around
everywhere (meaning another pass hopefully won't be needed).

Go for the most common style:

 - Indent properties with a single tab, including for choices.

   Properties on choices work exactly the same syntactically as
   properties on symbols, so not sure how the no-indentation thing
   happened.

 - Indent help texts with a tab followed by two spaces

 - Put a space between 'config' and the symbol name, not a tab. This
   also helps when grepping for definitions.

 - Do '# A comment' instead of '#A comment'

I tweaked Kconfiglib a bit to find most of the stuff.

Some help texts were reflowed to 79 columns with 'gq' in Vim as well,
though not all, because I was afraid I'd accidentally mess up
formatting.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

pfalcon commented on arch/Kconfig:
```diff
@@ -69,15 +69,14 @@ module-str = mpu
-         line option for gen_isr_tables.py.
+	bool
+	help
+	  This option tells the build system that the target system is big-endian.
```
I don't see this kind of change - re-wrapping lines to a different width - to have been described in the commit message. What's the reason for it, and how it was done? E.g., does it try to establish a consistent line width (definitely worth describing in the commit message), or did you just take a chance to do something when you fixed indent there anyway?

> (meaning another pass hopefully won't be needed)

ulfalizer replied:
Used `gq` in Vim to wrap to 79 columns. Only did it for a few of them though, because I was afraid I'd mess up some formatting while batch-fixing.

ulfalizer replied:
Commit message now mentions it.

pfalcon replied:
ack, thanks
